src/types/api.ts

```diff
@@ -11,3 +11,28 @@ export interface AuthResponse {
 export interface Token {
   token: string;
 }
+
+export interface APIArrayResponse<T> {
+  results: T[];
+}
+
+export interface APIResponse<T> {
+  data?: T;
+}
+
+export enum APIProgress {
+  INIT = 'INIT',
+  REQ = 'REQ',
+  ERR = 'ERR',
+  SUCCESS = 'SUCCESS',
+}
+
+export interface APIState {
+  error?: string;
+  progress: APIProgress;
+}
+
+export const INITIAL_API_STATE: APIState = {
```

Comment:
Hi @thesanjeevsharma, I created types `APIState` that keeps track of the state of API calls, which would be useful if we want to know if api calls are still undergoing or successful (then we can use this state to show the loader or the useful content). Does this look ok to u? 

```
try {
    setAPIState({...apiState, progress: APIProgress.REQ});
    const coreTeams = await teamsApi.getCoreTeams();
    setAPIState({...apiState, progress: APIProgress.SUCCESS});
    setTeams(coreTeams);
} catch (err) {
    setAPIState({error: err, progress: APIProgress.ERR});
}
```

Reply:
Nice, looks more organized. But isn't this a bit of overkill for showing a loader? Did you check earlier implementations?

@angle943 should we follow this?

Reply:
The earlier implementations just uses a local `isLoading` state to keep track. 

Hmm true, might be a bit of overkill for our use case. Let me think

Reply:
Actually, I believe `APIState` is useful for the following cases, where you have more control to decide what content to display:

- when the api hasn't been called yet (you don't want to display the content before the api is called) 
- when the calling of api encounters an error

Reply:
yeah this is good. quesiton, what is `REQ`? requesting?

Reply:
yes `REQ` stands for requesting